test/integration/types_test.rb

```diff
@@ -268,6 +268,71 @@ def test_escaped_data
     end
   end
 
+  def test_copy_data_in_fixed_size_binary_column
+    # test for the BINARY columns needing 0-byte padding
+    #
+    # For details, see https://github.com/Shopify/ghostferry/pull/159
+
+    [source_db, target_db].each do |db|
+      db.query("CREATE DATABASE IF NOT EXISTS #{DEFAULT_DB}")
+      db.query("CREATE TABLE IF NOT EXISTS #{DEFAULT_FULL_TABLE_NAME} (id bigint(20) not null auto_increment, data BINARY(4), primary key(id))")
+    end
+
+    # NOTE: We explicitly test with trailing 0s, because the MySQL replication
+    # master will strip such trailing 0s when streaming events to us. As a
+    # result, the binlog writer must explicitly add then when building
+    # update/delete statements, as the WHERE clause would not match existing
+    # rows in the target DB
+    inserted_data = "ABC\x00"
+    updated_data = "EFGH"
+
+    source_db.query("INSERT INTO #{DEFAULT_FULL_TABLE_NAME} (id, data) VALUES (1, _binary'#{inserted_data}')")
```

Comment:
should this be inserting a value that is shorter than the column definition(eg: `"ABC"`) and let mysql add the trailing 0s to reproduce the exact same issue reported here ?

Reply:
interestingly mysql seems to even strip the trailing whitespace if done like this. But your suggestion is perfectly valid and maybe a slightly better test-case